trio/_core/_run.py

```diff
@@ -13,7 +13,11 @@
 from contextvars import copy_context
 from math import inf
 from time import perf_counter
-from typing import Callable, TYPE_CHECKING
+from typing import Callable, NoReturn, TypeVar, TYPE_CHECKING
+from typing import Deque
```

Comment:
This should be a single import

Reply:
Also quick note that `typing.Callable` is deprecated, even though the alternative is 3.9+ which is awkward.

Reply:
Sorry for taking a long time to get back to you. Thank you for reviewing this. These issues should be addressed now.

Reply:
`typing.Deque` is deprecated too, as `collections.deque` is generic now.

Reply:
In the deque case, I'd rather not do anything until python 3.8 EOL since the type identifiers aren't exactly the same. I'd have to either do `from typing import Deque as deque` or `from collections.abc import deque as Deque`, both of which look kinda ugly tbh.

Reply:
While trio still promises support for python versions  < 3.9, using `deque` as a generic type will lead to runtime errors, unless we also do `from __future__ import annotations`.

Typing with `Deque` allows for backwards compatibility until support for python 3.8 and older is discontinued.

Reply:
Well, what's the problem with `from __future__ import annotations`? AnyIO uses it everywhere and so far nobody has complained.

Reply:
It makes some runtime uses of annotations more difficult. But I doubt that'll be an issue here. I can go with that approach.